aiida/restapi/translator/calculation/__init__.py

```diff
@@ -38,3 +40,117 @@ def __init__(self, **kwargs):
         super(CalculationTranslator, self).__init__(
             Class=self.__class__, **kwargs)
 
+    @staticmethod
+    def get_retrieved_inputs(node, filename=None, rtype=None):
+        """
+        Get the retrieved input files for job calculation
+        :param node: aiida node
+        :return: the retrieved input files for job calculation
+        """
+
+        if node.dbnode.type.startswith("calculation.job."):
+
+            fullpath = os.path.join(node.get_abs_path(), "raw_input")
+
+            if filename is not None:
+                if rtype is None:
+                    rtype = "download"
+
+                response = {}
+                if rtype == "download":
+                    try:
+                        filepath = os.path.join(fullpath, filename)
+                        dirpath = os.path.dirname(filepath)
+                        fname = os.path.basename(filepath)
+
+                        if os.path.isfile(filepath):
+                            response["status"] = 200
+                            response["data"] = send_from_directory(dirpath, fname)
+                            response["filename"] = filename.replace("/", "_")
+                        else:
+                            response["status"] = 500
+                            response["data"] = "file {} does not exist".format(filename)
+                    except Exception as e:
+                        response["status"] = 500
+                        response["data"] = e.message
+                else:
+                    response["status"] = 500
+                    response["data"] = "type is not supported"
+
+                return response
+
+            # if filename is not provided, return list of all retrieved files
+            try:
+                length = len(fullpath) + 1
+                retrieved = []
+
+                for (dirpath, dirnames, filenames) in os.walk(fullpath):
+                    retrieved.extend([os.path.join(dirpath, f)[length:] for f in filenames])
+
+                return retrieved
+
+            except Exception as e:
+                return e
```

Comment:
If there is an exception, is it ok to return it? It gets 'swallowed', and I don't think it can be serialised to JSON, so probably you can an error elsewhere? Or you have a specific management of the return value depending on whether it's an exception? But maybe in this case it's better to just not do anything here, and wrap the caller function in a `try...except`?

Reply:
there is a error handler written to return the exceptions.

<pre>
           @self.errorhandler(Exception)
            def error_handler(error):

                if isinstance(error, RestValidationError):
                    response = jsonify({'message': error.message})
                    response.status_code = 400

                elif isinstance(error, RestInputValidationError):
                    response = jsonify({'message': error.message})
                    response.status_code = 400

                elif isinstance(error, RestFeatureNotAvailable):
                    response = jsonify({'message': error.message})
                    response.status_code = 501

                # Generic server-side error (not to make the api crash if an
                # unhandled exception is raised. Caution is never enough!!)
                else:
                    response = jsonify(
                        {
                            'message': 'Internal server error. The original '
                                       'message was: \"{}\"'.format(
                                error.message)
                        }
                    )
                    response.status_code = 500

                return response
</pre>

In this case, I will return response with status 500 and error message. Is that ok?

Reply:
Ok I think this is ok if it already worked like this!